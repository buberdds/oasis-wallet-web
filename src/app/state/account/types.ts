import { Transaction } from 'app/state/transaction/types'
import { ErrorPayload } from 'types/errors'
import { StringifiedBigInt } from 'types/StringifiedBigInt'

export interface BalanceDetails {
  available: StringifiedBigInt | null
  debonding: StringifiedBigInt | null
  delegations: StringifiedBigInt | null
  total: StringifiedBigInt | null
}

export interface Allowance {
  address: string
  amount: StringifiedBigInt
}

export interface Account extends BalanceDetails {
  address: string
  allowances?: Allowance[]
}

/* --- STATE --- */
export interface AccountState extends Account {
  loading: boolean
  accountError?: ErrorPayload
  transactions: Transaction[]
  transactionsError?: ErrorPayload
}
