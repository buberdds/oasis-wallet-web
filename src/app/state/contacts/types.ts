export interface Contact {
  address: string
  name: string
}

/* --- STATE --- */
export interface ContactsState {
  [key: string]: Contact
}
