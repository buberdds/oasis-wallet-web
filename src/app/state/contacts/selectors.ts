import { createSelector } from '@reduxjs/toolkit'

import { RootState } from 'types'
import { initialState } from '.'

const selectSlice = (state: RootState) => state.contacts || initialState

export const selectContactsList = createSelector([selectSlice], contacts =>
  Object.keys(contacts).map(contact => ({
    address: contacts[contact].address,
    name: contacts[contact].name,
  })),
)

export const selectContact = (address: string) => createSelector([selectSlice], contacts => contacts[address])
