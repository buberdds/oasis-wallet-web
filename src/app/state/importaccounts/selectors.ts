import { createSelector } from '@reduxjs/toolkit'

import { RootState } from 'types'
import { initialState } from '.'

const selectSlice = (state: RootState) => state.importAccounts || initialState

export const selectError = createSelector([selectSlice], state => state.error)
export const selectImportAccounts = createSelector([selectSlice], state => state)
export const selectImportAccountsList = createSelector([selectSlice], state => state.accounts)
export const selectDeviceRequested = createSelector([selectSlice], state => state.deviceRequested)
export const selectShowAccountsSelectionModal = createSelector(
  [selectSlice],
  state => state.showAccountsSelectionModal,
)
export const selectSelectedAccounts = createSelector([selectImportAccountsList], state =>
  state.filter(a => a.selected),
)
