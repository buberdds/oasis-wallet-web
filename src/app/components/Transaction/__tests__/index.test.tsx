import { render, screen } from '@testing-library/react'
import userEvent from '@testing-library/user-event'
import { selectTicker } from 'app/state/network/selectors'
import { when } from 'jest-when'
import * as React from 'react'
import { Router } from 'react-router'
import { createMemoryHistory } from 'history'
import { Provider, useSelector } from 'react-redux'
import { mocked } from 'ts-jest/utils'
import { configureAppStore } from 'store/configureStore'
import { BackendAPIs, backend } from 'vendors/backend'

import { Transaction } from '..'
import * as transactionTypes from 'app/state/transaction/types'
import { NetworkType } from 'app/state/network/types'

jest.mock('vendors/backend')
jest.mock('react-redux', () => ({
  ...jest.requireActual('react-redux'),
  useSelector: jest.fn(),
}))

const history = createMemoryHistory()
const pushSpy = jest.spyOn(history, 'push')
const renderComponent = (
  store: any,
  ref: any,
  transaction: transactionTypes.Transaction,
  network: NetworkType,
) =>
  render(
    <Router history={history}>
      <Provider store={store}>
        <Transaction referenceAddress={ref} transaction={transaction} network={network} />
      </Provider>
    </Router>,
  )

describe('<Transaction  />', () => {
  let store: ReturnType<typeof configureAppStore>
  const ref = 'sourceAddr'
  const transaction = {
    amount: 1000000,
    timestamp: 1618018255,
    from: 'source',
    to: 'destination',
    type: transactionTypes.TransactionType.StakingTransfer,
    hash: 'ff1234',
  } as transactionTypes.Transaction
  const network = 'mainnet'

  beforeEach(() => {
    mocked(backend).mockImplementation(() => BackendAPIs.OasisScan)
    store = configureAppStore()

    when(useSelector as any)
      .calledWith(selectTicker)
      .mockReturnValue('TEST')
  })

  it('should match snapshot', () => {
    const component = renderComponent(store, ref, transaction, network)
    expect(component.container.firstChild).toMatchSnapshot()
  })

  it('should redirect user when clicking on address section', () => {
    renderComponent(store, ref, transaction, network)

    userEvent.click(screen.getByLabelText('ContactInfo'))
    expect(pushSpy).toHaveBeenCalledWith(
      expect.objectContaining({
        pathname: `/account/${transaction.from}`,
      }),
    )
  })

  it('should not redirect user when clicking on amount or block section', () => {
    renderComponent(store, ref, transaction, network)

    userEvent.click(screen.getByLabelText('Money'))
    userEvent.click(screen.getByLabelText('Cube'))
    expect(pushSpy).not.toHaveBeenCalled()
  })

  it('should handle unknown transaction types gracefully', () => {
    const component = renderComponent(
      store,
      'sourceAddr',
      {
        amount: 1000000,
        timestamp: 1618018255,
        from: 'source',
        to: 'destination',
        type: 'turboencabulate' as transactionTypes.TransactionType,
        hash: 'ff1234',
        fee: undefined,
        level: undefined,
        status: true,
      },
      network,
    )
    expect(component.container.firstChild).toMatchSnapshot()
  })

  it('should not render a link when address is undefined', () => {
    renderComponent(
      store,
      'sourceAddr',
      {
        amount: 1000000,
        timestamp: 1618018255,
        from: 'sourceAddr',
        to: undefined,
        type: 'anyType' as transactionTypes.TransactionType,
        hash: 'ff1234',
        fee: undefined,
        level: undefined,
        status: true,
      },
      network,
    )

    expect(screen.queryByTestId('external-wallet-address')).not.toBeInTheDocument()
    expect(screen.getByText('common.unavailable')).toBeInTheDocument()
  })

  it('should render testnet link', () => {
    renderComponent(store, ref, transaction, 'testnet')
    expect(screen.getByTestId('explorer-link')).toHaveAttribute(
      'href',
      'https://testnet.oasisscan.com/transactions/ff1234',
    )
  })

  it('should render monitor link', () => {
    mocked(backend).mockImplementation(() => BackendAPIs.OasisMonitor)
    renderComponent(store, ref, transaction, network)
    expect(screen.getByTestId('explorer-link')).toHaveAttribute(
      'href',
      'https://oasismonitor.com/operation/ff1234',
    )
  })

  it('should render testnet monitor link', () => {
    mocked(backend).mockImplementation(() => BackendAPIs.OasisMonitor)
    renderComponent(store, ref, transaction, 'testnet')
    expect(screen.getByTestId('explorer-link')).toHaveAttribute(
      'href',
      'https://testnet.oasismonitor.com/operation/ff1234',
    )
  })
})
