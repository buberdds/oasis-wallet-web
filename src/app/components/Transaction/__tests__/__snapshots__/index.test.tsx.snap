// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Transaction  /> should handle unknown transaction types gracefully 1`] = `
.c3 {
  display: inline-block;
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
  width: 24px;
  height: 24px;
  fill: #f8f8f8;
  stroke: #f8f8f8;
}

.c3 g {
  fill: inherit;
  stroke: inherit;
}

.c3 *:not([stroke])[fill="none"] {
  stroke-width: 0;
}

.c3 *[stroke*="#"],
.c3 *[STROKE*="#"] {
  stroke: inherit;
  fill: none;
}

.c3 *[fill-rule],
.c3 *[FILL-RULE],
.c3 *[fill*="#"],
.c3 *[FILL*="#"] {
  fill: inherit;
  stroke: none;
}

.c10 {
  display: inline-block;
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
  width: 24px;
  height: 24px;
  fill: #7D4CDB;
  stroke: #7D4CDB;
}

.c10 g {
  fill: inherit;
  stroke: inherit;
}

.c10 *:not([stroke])[fill="none"] {
  stroke-width: 0;
}

.c10 *[stroke*="#"],
.c10 *[STROKE*="#"] {
  stroke: inherit;
  fill: none;
}

.c10 *[fill-rule],
.c10 *[FILL-RULE],
.c10 *[fill*="#"],
.c10 *[FILL*="#"] {
  fill: inherit;
  stroke: none;
}

.c17 {
  display: inline-block;
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
  width: 24px;
  height: 24px;
  fill: #777777;
  stroke: #777777;
}

.c17 g {
  fill: inherit;
  stroke: inherit;
}

.c17 *:not([stroke])[fill="none"] {
  stroke-width: 0;
}

.c17 *[stroke*="#"],
.c17 *[STROKE*="#"] {
  stroke: inherit;
  fill: none;
}

.c17 *[fill-rule],
.c17 *[FILL-RULE],
.c17 *[fill*="#"],
.c17 *[FILL*="#"] {
  fill: inherit;
  stroke: none;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  box-sizing: border-box;
  max-width: 100%;
  background-color: #FFFFFF;
  color: #444444;
  min-width: 0;
  min-height: 0;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  border-radius: 12px;
  overflow: hidden;
  box-shadow: 0px 1px 2px rgba(0,0,0,0.20);
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  box-sizing: border-box;
  max-width: 100%;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background-color: #7D4CDB;
  color: #f8f8f8;
  min-width: 0;
  min-height: 0;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  padding-left: 24px;
  padding-right: 24px;
  padding-top: 12px;
  padding-bottom: 12px;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  box-sizing: border-box;
  max-width: 100%;
  min-width: 0;
  min-height: 0;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
}

.c6 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  box-sizing: border-box;
  max-width: 100%;
  min-width: 0;
  min-height: 0;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-flex: 1 1;
  -ms-flex: 1 1;
  flex: 1 1;
  padding: 0px;
}

.c8 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  box-sizing: border-box;
  max-width: 100%;
  min-width: 0;
  min-height: 0;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  padding: 24px;
}

.c9 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  box-sizing: border-box;
  max-width: 100%;
  -webkit-align-self: center;
  -ms-flex-item-align: center;
  align-self: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  min-width: 0;
  min-height: 0;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  height: 100%;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  padding-right: 6px;
}

.c11 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  box-sizing: border-box;
  max-width: 100%;
  min-width: 0;
  min-height: 0;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c13 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  box-sizing: border-box;
  max-width: 100%;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background: #33333310;
  min-width: 0;
  min-height: 0;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  padding-left: 24px;
  padding-right: 24px;
}

.c4 {
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
  -webkit-align-self: stretch;
  -ms-flex-item-align: stretch;
  align-self: stretch;
  width: 12px;
}

.c15 {
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
  -webkit-align-self: stretch;
  -ms-flex-item-align: stretch;
  align-self: stretch;
  width: 24px;
}

.c5 {
  font-size: 18px;
  line-height: 24px;
}

.c12 {
  font-size: 18px;
  line-height: 24px;
  font-weight: bold;
}

.c14 {
  font-size: 14px;
  line-height: 20px;
}

.c16 {
  display: inline-block;
  box-sizing: border-box;
  cursor: pointer;
  font: inherit;
  -webkit-text-decoration: none;
  text-decoration: none;
  margin: 0;
  background: transparent;
  overflow: visible;
  text-transform: none;
  color: inherit;
  outline: none;
  border: none;
  padding: 0;
  text-align: inherit;
  line-height: 0;
  padding: 12px;
}

.c16:hover {
  background-color: rgba(221,221,221,0.4);
  color: #000000;
}

.c16:focus {
  outline: none;
  box-shadow: 0 0 2px 2px #6FFFB0;
}

.c16:focus > circle,
.c16:focus > ellipse,
.c16:focus > line,
.c16:focus > path,
.c16:focus > polygon,
.c16:focus > polyline,
.c16:focus > rect {
  outline: none;
  box-shadow: 0 0 2px 2px #6FFFB0;
}

.c16:focus::-moz-focus-inner {
  border: 0;
}

.c16:focus:not(:focus-visible) {
  outline: none;
  box-shadow: none;
}

.c16:focus:not(:focus-visible) > circle,
.c16:focus:not(:focus-visible) > ellipse,
.c16:focus:not(:focus-visible) > line,
.c16:focus:not(:focus-visible) > path,
.c16:focus:not(:focus-visible) > polygon,
.c16:focus:not(:focus-visible) > polyline,
.c16:focus:not(:focus-visible) > rect {
  outline: none;
  box-shadow: none;
}

.c16:focus:not(:focus-visible)::-moz-focus-inner {
  border: 0;
}

.c7 {
  display: grid;
  box-sizing: border-box;
  grid-template-columns: repeat( auto-fit,minmax(min(96px,100%),1fr) );
  grid-gap: 0px 0px;
}

@media only screen and (max-width:768px) {
  .c0 {
    border-radius: 6px;
  }
}

@media only screen and (max-width:768px) {
  .c1 {
    padding-left: 12px;
    padding-right: 12px;
  }
}

@media only screen and (max-width:768px) {
  .c1 {
    padding-top: 6px;
    padding-bottom: 6px;
  }
}

@media only screen and (max-width:768px) {
  .c6 {
    padding: 0px;
  }
}

@media only screen and (max-width:768px) {
  .c8 {
    padding: 12px;
  }
}

@media only screen and (max-width:768px) {
  .c9 {
    padding-right: 3px;
  }
}

@media only screen and (max-width:768px) {
  .c13 {
    padding-left: 12px;
    padding-right: 12px;
  }
}

@media only screen and (max-width:768px) {
  .c4 {
    width: 6px;
  }
}

@media only screen and (max-width:768px) {
  .c15 {
    width: 12px;
  }
}

<div
  class="c0"
>
  <header
    class="c1"
  >
    <div
      class="c2"
    >
      <svg
        aria-label="New"
        class="c3"
        viewBox="0 0 24 24"
      >
        <path
          d="M12 1v22M2 6l20 12m0-12L2 18"
          fill="none"
          stroke="#000"
          stroke-width="2"
        />
      </svg>
      <div
        class="c4"
      />
      <span
        class="c5"
      >
        account.otherTransaction.header
      </span>
    </div>
  </header>
  <div
    class="c6"
  >
    <div
      class="c7"
    >
      <div
        class="c8"
      >
        <div
          class="c9"
        >
          <svg
            aria-label="Money"
            class="c10"
            viewBox="0 0 24 24"
          >
            <path
              d="M16 16c0-1.105-3.134-2-7-2s-7 .895-7 2 3.134 2 7 2 7-.895 7-2zM2 16v4.937C2 22.077 5.134 23 9 23s7-.924 7-2.063V16M9 5c-4.418 0-8 .895-8 2s3.582 2 8 2M1 7v5c0 1.013 3.582 2 8 2M23 4c0-1.105-3.1-2-6.923-2-3.824 0-6.923.895-6.923 2s3.1 2 6.923 2S23 5.105 23 4zm-7 12c3.824 0 7-.987 7-2V4M9.154 4v10.166M9 9c0 1.013 3.253 2 7.077 2C19.9 11 23 10.013 23 9"
              fill="none"
              stroke="#000"
              stroke-width="2"
            />
          </svg>
        </div>
        <div
          class="c4"
        />
        <div
          class="c11"
        >
          <span
            class="c12"
          >
            common.amount
          </span>
          <span
            class="c5"
          >
            0.001
             
            TEST
          </span>
        </div>
      </div>
      <a
        data-testid="external-wallet-address"
        href="/account/source"
      >
        <div
          class="c8"
        >
          <div
            class="c9"
          >
            <svg
              aria-label="ContactInfo"
              class="c10"
              viewBox="0 0 24 24"
            >
              <path
                d="M5 12a3 3 0 1 1 0-6 3 3 0 0 1 0 6zm4 6v-2c0-2.25-1.787-4-4.036-4h.054C2.768 12 1 13.75 1 16v2M12 7h12M12 17h10m-10-5h7"
                fill="none"
                stroke="#000"
                stroke-width="2"
              />
            </svg>
          </div>
          <div
            class="c4"
          />
          <div
            class="c11"
          >
            <span
              class="c12"
            >
              account.otherTransaction.designation
            </span>
            <span
              class="c5"
            >
              source...source
            </span>
          </div>
        </div>
      </a>
      <div
        class="c8"
      >
        <div
          class="c9"
        >
          <svg
            aria-label="Cube"
            class="c10"
            viewBox="0 0 24 24"
          >
            <path
              d="m12 2 10 5v10l-10 5-10-5V7l10-5zM2 7l10 5 10-5m-10 5v10-10z"
              fill="none"
              stroke="#000"
              stroke-width="2"
            />
          </svg>
        </div>
        <div
          class="c4"
        />
        <div
          class="c11"
        >
          <span
            class="c12"
          >
            common.block
          </span>
          <span
            class="c5"
          />
        </div>
      </div>
    </div>
  </div>
  <footer
    class="c13"
  >
    <span
      class="c14"
    >
      10/04/2021, 01:30:55
    </span>
    <div
      class="c15"
    />
    <div
      class="c2"
    >
      <a
        class="c16"
        data-testid="explorer-link"
        href="https://oasisscan.com/transactions/ff1234"
        rel="noopener"
        target="_blank"
      >
        <svg
          aria-label="CircleInformation"
          class="c17"
          viewBox="0 0 24 24"
        >
          <path
            d="M12 22c5.523 0 10-4.477 10-10S17.523 2 12 2 2 6.477 2 12s4.477 10 10 10zm0-12v8m0-12v2"
            fill="none"
            stroke="#000"
            stroke-width="2"
          />
        </svg>
      </a>
    </div>
  </footer>
</div>
`;

exports[`<Transaction  /> should match snapshot 1`] = `
.c3 {
  display: inline-block;
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
  width: 24px;
  height: 24px;
  fill: #f8f8f8;
  stroke: #f8f8f8;
}

.c3 g {
  fill: inherit;
  stroke: inherit;
}

.c3 *:not([stroke])[fill="none"] {
  stroke-width: 0;
}

.c3 *[stroke*="#"],
.c3 *[STROKE*="#"] {
  stroke: inherit;
  fill: none;
}

.c3 *[fill-rule],
.c3 *[FILL-RULE],
.c3 *[fill*="#"],
.c3 *[FILL*="#"] {
  fill: inherit;
  stroke: none;
}

.c10 {
  display: inline-block;
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
  width: 24px;
  height: 24px;
  fill: #7D4CDB;
  stroke: #7D4CDB;
}

.c10 g {
  fill: inherit;
  stroke: inherit;
}

.c10 *:not([stroke])[fill="none"] {
  stroke-width: 0;
}

.c10 *[stroke*="#"],
.c10 *[STROKE*="#"] {
  stroke: inherit;
  fill: none;
}

.c10 *[fill-rule],
.c10 *[FILL-RULE],
.c10 *[fill*="#"],
.c10 *[FILL*="#"] {
  fill: inherit;
  stroke: none;
}

.c17 {
  display: inline-block;
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
  width: 24px;
  height: 24px;
  fill: #777777;
  stroke: #777777;
}

.c17 g {
  fill: inherit;
  stroke: inherit;
}

.c17 *:not([stroke])[fill="none"] {
  stroke-width: 0;
}

.c17 *[stroke*="#"],
.c17 *[STROKE*="#"] {
  stroke: inherit;
  fill: none;
}

.c17 *[fill-rule],
.c17 *[FILL-RULE],
.c17 *[fill*="#"],
.c17 *[FILL*="#"] {
  fill: inherit;
  stroke: none;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  box-sizing: border-box;
  max-width: 100%;
  background-color: #FFFFFF;
  color: #444444;
  min-width: 0;
  min-height: 0;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  border-radius: 12px;
  overflow: hidden;
  box-shadow: 0px 1px 2px rgba(0,0,0,0.20);
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  box-sizing: border-box;
  max-width: 100%;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background-color: #7D4CDB;
  color: #f8f8f8;
  min-width: 0;
  min-height: 0;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  padding-left: 24px;
  padding-right: 24px;
  padding-top: 12px;
  padding-bottom: 12px;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  box-sizing: border-box;
  max-width: 100%;
  min-width: 0;
  min-height: 0;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
}

.c6 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  box-sizing: border-box;
  max-width: 100%;
  min-width: 0;
  min-height: 0;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-flex: 1 1;
  -ms-flex: 1 1;
  flex: 1 1;
  padding: 0px;
}

.c8 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  box-sizing: border-box;
  max-width: 100%;
  min-width: 0;
  min-height: 0;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  padding: 24px;
}

.c9 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  box-sizing: border-box;
  max-width: 100%;
  -webkit-align-self: center;
  -ms-flex-item-align: center;
  align-self: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  min-width: 0;
  min-height: 0;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  height: 100%;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  padding-right: 6px;
}

.c11 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  box-sizing: border-box;
  max-width: 100%;
  min-width: 0;
  min-height: 0;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c13 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  box-sizing: border-box;
  max-width: 100%;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background: #33333310;
  min-width: 0;
  min-height: 0;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  padding-left: 24px;
  padding-right: 24px;
}

.c4 {
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
  -webkit-align-self: stretch;
  -ms-flex-item-align: stretch;
  align-self: stretch;
  width: 12px;
}

.c15 {
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
  -webkit-align-self: stretch;
  -ms-flex-item-align: stretch;
  align-self: stretch;
  width: 24px;
}

.c5 {
  font-size: 18px;
  line-height: 24px;
}

.c12 {
  font-size: 18px;
  line-height: 24px;
  font-weight: bold;
}

.c14 {
  font-size: 14px;
  line-height: 20px;
}

.c16 {
  display: inline-block;
  box-sizing: border-box;
  cursor: pointer;
  font: inherit;
  -webkit-text-decoration: none;
  text-decoration: none;
  margin: 0;
  background: transparent;
  overflow: visible;
  text-transform: none;
  color: inherit;
  outline: none;
  border: none;
  padding: 0;
  text-align: inherit;
  line-height: 0;
  padding: 12px;
}

.c16:hover {
  background-color: rgba(221,221,221,0.4);
  color: #000000;
}

.c16:focus {
  outline: none;
  box-shadow: 0 0 2px 2px #6FFFB0;
}

.c16:focus > circle,
.c16:focus > ellipse,
.c16:focus > line,
.c16:focus > path,
.c16:focus > polygon,
.c16:focus > polyline,
.c16:focus > rect {
  outline: none;
  box-shadow: 0 0 2px 2px #6FFFB0;
}

.c16:focus::-moz-focus-inner {
  border: 0;
}

.c16:focus:not(:focus-visible) {
  outline: none;
  box-shadow: none;
}

.c16:focus:not(:focus-visible) > circle,
.c16:focus:not(:focus-visible) > ellipse,
.c16:focus:not(:focus-visible) > line,
.c16:focus:not(:focus-visible) > path,
.c16:focus:not(:focus-visible) > polygon,
.c16:focus:not(:focus-visible) > polyline,
.c16:focus:not(:focus-visible) > rect {
  outline: none;
  box-shadow: none;
}

.c16:focus:not(:focus-visible)::-moz-focus-inner {
  border: 0;
}

.c7 {
  display: grid;
  box-sizing: border-box;
  grid-template-columns: repeat( auto-fit,minmax(min(96px,100%),1fr) );
  grid-gap: 0px 0px;
}

@media only screen and (max-width:768px) {
  .c0 {
    border-radius: 6px;
  }
}

@media only screen and (max-width:768px) {
  .c1 {
    padding-left: 12px;
    padding-right: 12px;
  }
}

@media only screen and (max-width:768px) {
  .c1 {
    padding-top: 6px;
    padding-bottom: 6px;
  }
}

@media only screen and (max-width:768px) {
  .c6 {
    padding: 0px;
  }
}

@media only screen and (max-width:768px) {
  .c8 {
    padding: 12px;
  }
}

@media only screen and (max-width:768px) {
  .c9 {
    padding-right: 3px;
  }
}

@media only screen and (max-width:768px) {
  .c13 {
    padding-left: 12px;
    padding-right: 12px;
  }
}

@media only screen and (max-width:768px) {
  .c4 {
    width: 6px;
  }
}

@media only screen and (max-width:768px) {
  .c15 {
    width: 12px;
  }
}

<div
  class="c0"
>
  <header
    class="c1"
  >
    <div
      class="c2"
    >
      <svg
        aria-label="LinkPrevious"
        class="c3"
        viewBox="0 0 24 24"
      >
        <path
          d="M22 12H2m9-9-9 9 9 9"
          fill="none"
          stroke="#000"
          stroke-width="2"
        />
      </svg>
      <div
        class="c4"
      />
      <span
        class="c5"
      />
    </div>
  </header>
  <div
    class="c6"
  >
    <div
      class="c7"
    >
      <div
        class="c8"
      >
        <div
          class="c9"
        >
          <svg
            aria-label="Money"
            class="c10"
            viewBox="0 0 24 24"
          >
            <path
              d="M16 16c0-1.105-3.134-2-7-2s-7 .895-7 2 3.134 2 7 2 7-.895 7-2zM2 16v4.937C2 22.077 5.134 23 9 23s7-.924 7-2.063V16M9 5c-4.418 0-8 .895-8 2s3.582 2 8 2M1 7v5c0 1.013 3.582 2 8 2M23 4c0-1.105-3.1-2-6.923-2-3.824 0-6.923.895-6.923 2s3.1 2 6.923 2S23 5.105 23 4zm-7 12c3.824 0 7-.987 7-2V4M9.154 4v10.166M9 9c0 1.013 3.253 2 7.077 2C19.9 11 23 10.013 23 9"
              fill="none"
              stroke="#000"
              stroke-width="2"
            />
          </svg>
        </div>
        <div
          class="c4"
        />
        <div
          class="c11"
        >
          <span
            class="c12"
          >
            common.amount
          </span>
          <span
            class="c5"
          >
            0.001
             
            TEST
          </span>
        </div>
      </div>
      <a
        data-testid="external-wallet-address"
        href="/account/source"
      >
        <div
          class="c8"
        >
          <div
            class="c9"
          >
            <svg
              aria-label="ContactInfo"
              class="c10"
              viewBox="0 0 24 24"
            >
              <path
                d="M5 12a3 3 0 1 1 0-6 3 3 0 0 1 0 6zm4 6v-2c0-2.25-1.787-4-4.036-4h.054C2.768 12 1 13.75 1 16v2M12 7h12M12 17h10m-10-5h7"
                fill="none"
                stroke="#000"
                stroke-width="2"
              />
            </svg>
          </div>
          <div
            class="c4"
          />
          <div
            class="c11"
          >
            <span
              class="c12"
            >
              common.from
            </span>
            <span
              class="c5"
            >
              source...source
            </span>
          </div>
        </div>
      </a>
      <div
        class="c8"
      >
        <div
          class="c9"
        >
          <svg
            aria-label="Cube"
            class="c10"
            viewBox="0 0 24 24"
          >
            <path
              d="m12 2 10 5v10l-10 5-10-5V7l10-5zM2 7l10 5 10-5m-10 5v10-10z"
              fill="none"
              stroke="#000"
              stroke-width="2"
            />
          </svg>
        </div>
        <div
          class="c4"
        />
        <div
          class="c11"
        >
          <span
            class="c12"
          >
            common.block
          </span>
          <span
            class="c5"
          />
        </div>
      </div>
    </div>
  </div>
  <footer
    class="c13"
  >
    <span
      class="c14"
    >
      10/04/2021, 01:30:55
    </span>
    <div
      class="c15"
    />
    <div
      class="c2"
    >
      <a
        class="c16"
        data-testid="explorer-link"
        href="https://oasisscan.com/transactions/ff1234"
        rel="noopener"
        target="_blank"
      >
        <svg
          aria-label="CircleInformation"
          class="c17"
          viewBox="0 0 24 24"
        >
          <path
            d="M12 22c5.523 0 10-4.477 10-10S17.523 2 12 2 2 6.477 2 12s4.477 10 10 10zm0-12v8m0-12v2"
            fill="none"
            stroke="#000"
            stroke-width="2"
          />
        </svg>
      </a>
    </div>
  </footer>
</div>
`;
